src/Reflection/InitializerExprTypeResolver.php

```diff
@@ -856,6 +856,10 @@ public function getModType(Expr $left, Expr $right, callable $getTypeCallback):
 			return $this->getNeverType($leftType, $rightType);
 		}
 
+		if ($leftType->toNumber() instanceof ErrorType || $rightType->toNumber() instanceof ErrorType) {
```

Comment:
I'm not sure whether this should only be enabled in bleeding edge. On the one hand, I don't see any other case in this class where `return new ErrorType` would be conditioned by bleeding edge. On the other hand, this can definitely result in new errors being reported without bleeding edge. E.g. `5 % []` now reports `Binary operation "%" between 5 and array{} results in an error`.

Reply:
The `$int % $string` use-case is even more compelling example of why it should apply only to bleeding edge.